libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -962,8 +964,13 @@ public void onMediaUploadSucceeded(final String localMediaId, final MediaFile me
 
     @Override
     public void onMediaUploadProgress(final String localMediaId, final float progress) {
-        mUploadingMediaProgressMax.put(localMediaId, progress);
-        getGutenbergContainerFragment().mediaFileUploadProgress(Integer.valueOf(localMediaId), progress);
+        if(!mCancelledMediaIds.contains(localMediaId)) {
+            mUploadingMediaProgressMax.put(localMediaId, progress);
+            getGutenbergContainerFragment().mediaFileUploadProgress(Integer.valueOf(localMediaId), progress);
+        }
+        else {
+            mCancelledMediaIds.remove(localMediaId);
```

Comment:
I am wondering if this approach of removing the `localMediaId` once it's within the `mCancelledMediaIds` is appropriate. So far during my testing only only one `onMediaUploadProgress` event is fired after cancellation is requested via the event sent from the `EditPostActivity` to the `UploadService`. Let me know your thoughts on this. I did this to ensure that if the same image with the same local media id is being reuploaded this does not prevent the progress from being propagated to the editor.

Reply:
yeah I get what you mean. I was thinking if the event comes rapidly after the cancellation, then we can disregard it. But if it comes say, after 2+ seconds then we can assume with some level of confidence that  this may be treated as a "new attempt". So we'd have to record the time when the user cancelled it along with its id, instead of it being a HashSet it could be a HashMap that links the time of cancellation to the localMediaid as key, then when onMediaUploadProgress is called we'd check if this is contained in mCancelledMediaIds.keys, and if yes we check the time elapsed before deciding to remove or not.
A proper signal would be to check when the user inserted the image in the first place, but I'm not sure we have a signal for this through the bridge. It may be appropriate to add a bridge method, but it also may be overkill for a corner case bug anyway.
I think adding the time check adds extra security and is not that much code to add, without going for the full blown solution, wdyt?

Reply:
Awesome. This sounds like a really sound idea. Thanks, much @mzorz I did an implementation in f60b6fd so I would like your feedback on my approach.